src/dawn/engine_dawn.cc

```diff
@@ -879,8 +929,40 @@ Result EngineDawn::DoDrawRect(const DrawRectCommand* command) {
       0.0f,
       1.0f,
   };
-  render_pipeline->vertex_buffer = CreateBufferFromData(
-      *device_, vertexData, sizeof(vertexData), ::dawn::BufferUsageBit::Vertex);
+  render_pipeline->vertex_buffer.emplace_back(
+      CreateBufferFromData(*device_, vertexData, sizeof(vertexData),
+                           ::dawn::BufferUsageBit::Vertex));
```

Comment:
Seems like it would be invalid to call DRAW_RECT on a pipeline that has VERTEX_DATA attached to it.
I filed https://github.com/google/amber/issues/560 

(Both amberscript and vkscript docs say user-supplied vertex data or index data is for use with draw-arrays.)

Reply:
in AttachBuffersAndTextures we also create and add other buffers (storage, unifrom, index) so call it will redo all these and corrupts the layout.

I added a comment on #560.